### PR TITLE
Graziani 2021-11-blog

### PR DESCRIPTION
**Pull request by bernhold**
# Instructions (DELETE THIS SECTION)

To view these instructions, please hit the **Preview** tab (above left). After you have read all of these instructions, delete them entirely from the body of this issue before entering your submission.

Be sure to select `master` as the `base` branch above as the target for this PR.

Provide a general summary of your changes in the title above.

Fill in the below **Description** section with minimal text describing the changes/new contributions in this PR and replace `<...>` as appropriate.

Remove the beginning `<!--` and ending `-->` comment markers for the correct **checklist** to be used for this PR and delete the other checklist.  (The checklist items for files displayed on the bssw.io site are shown by default.  The checklist items for internal files not displayed in the bssw.io site is given below with commented out checklist items by default.)

Any checklist items that don't apply can be striked out by adding `~~` to the beginning and end of the checklist item as `* ~~[] <checklist-item>~~`.  Also, remove the strikeout markers `~~` for the [wikize_refs.py] checklist items if using formal citations for bssw.io contributions.

# Description

EB Member: @`<eb-member-id>`

Resolves #`<issue-id>`

`<Other minimal information about the PR.>`

## PR checklist for files displayed on bssw.io site

* [ ] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [ ] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py])~~.
* [ ] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [ ] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [ ] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [ ] ***[EB Mem]*** Add one or more Reviewers.
* [ ] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [ ] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [ ] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: yes`).
* [ ] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [ ] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [ ] ***[Author]*** Make any final changes to the PR based on feedback.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py]).~~
* [ ] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [ ] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [ ] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [ ] ***[EB Mem]*** Leave comment and @mention `@betterscientificsoftware/bssw-maint` (**BSSw Maint**) asking to carry out final publication steps.
* [ ] ***[BSSw Maint]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] ***[BSSw Maint]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [ ] ***[BSSw Maint]*** Verify new contribution shows up on [bssw.io] as expected.

NOTE:
* Checklist items prefixed with ***[Author]*** are expected to be performed by the author of the PR or can be performed by the author.
* Checklist items prefixed with ***[EB Mem]*** must be performed by a [BSSw.io Editorial Board (EB) Member](https://betterscientificsoftware.github.io/bssw.io/bssw_members.html).
* Checklist items prefixed with ***[BSSw Maint]*** must be performed by someone in the `@betterscientificsoftware/bssw-maint` team (hint: type `@`, `b`, `s`, `s`, `w`, `-`, `m`  to auto-complete to `@betterscientificsoftware/bssw-maint`).

## PR checklist for (internal) files not displayed on bssw.io site

*Click "Write" above and remove comment markers to see below checklist items*

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

**Comment by bernhold** [comment]
@markcmiller86 I could use some help with the references in this article.  When I run wikize_refs on it, I get 
> Some footnotes never appear in the references...
['3', '1', '2', '5', '4']

which doesn't seem right on its face.  However, in looking at your documentation, I see that the references are expected to conform to a particular structure involving a URL and a title, which these don't really follow.  They are more asides in the form of footnotes rather than references per se.  So maybe this is not what your tool was intended for.  If that's the case, its fine.  I just want to understand so I can figure out the best strategy to deal with them.  But if they should work, and I'm missing something, I'd *definitely* appreciate your help.  Thanks

**Comment by markcmiller86** [comment]
Can you try just following the advice it prints add ad `-w` option to command line...that turns failures into warnings and should proceed with producing adjusted `.md` file.

**Comment by markcmiller86** [comment]
FYI...the markdown for the footnotes in that file is wrong. It seems to be trying to use GFM *reference style* links as a shorthand for generic footnoting.

Reference style links are simply a different way of doing links avoiding having to inline the URLs. So, some inline text like `[foo](https://www.bar.com)` can be replaced with `[foo][bar]` and a separate line of the form`[bar]: https://www.bar.com` or `[foo]` and a separate line of the form `[foo]: https://www.bar.com`. So, those lines at the bottom of the current file with `[1]: xxxx...` *must* specify a URL in order for it to be valid markdown. They are not designed to handle arbitrary footnote text.

What this doc currently has for footnote 1 for example is equiv. to `[1](The beginnings of open-science HPC as a sustained activity&mdash;as distinct from classified science HPC conducted at US DOE national labs&mdash;can probably be dated to DOE's ASC Alliance program in the late 1990s.)` which we agree is incorrectly formatted markdown.

If all you want to do is generic footnote, one would not use GFM's reference style links to achieve that. You would remove the square brackets around all the footnote identifiers. So, something like...

```
This is a footnote<sup>1</sup> reference.

<sup>1</sup>This is the footnote to which the above reference refers.
```

Note that there is no *linking* going on there. Its just typographical footnoting. If you really wanted linking (e.g. the ref takes you to the actual footnote), you could do that but you'd either manually add the links and anchors to do that or utilize wikize_refs.py to help automate it for you by spoofing some links and anchors. I can try to produce an example of that if you'd like.

**Comment by markcmiller86** [comment]
This slight modification to the ref links section gets you 90% there...

```
[1]: # "The beginnings of open-science HPC as a sustained activity&mdash;as distinct from classified science HPC conducted at US DOE national labs&mdash;can probably be dated to DOE's ASC Alliance program in the late 1990s."
[2]: # "*Continuing to accomplish them for 25 years*&mdash;the FLASH code is a workhorse of computational fluid dynamics with an enormous user community to this day&mdash;well, that takes us beyond the narrative in this essay, but phrases such as 'tour de force' strike me as too weak to suit the case."
[3]: # "The coin of the realm."
[4]: # "Howard M. Kanare, *Writing the Laboratory Notebook* (American Chemical Society, Washington, D.C., ISBN-0-8412-0933-2), 1985. <https://files.eric.ed.gov/fulltext/ED344734.pdf>"
[5]: # "If&mdash;ok, fine, when&mdash;the field of social psychology succeeds in dragging itself out of the [Reproducibility Crisis](https://www.wired.com/story/social-science-reproducibility/), I wouldn't be at all surprised if it turned out that a large part of the credit will be given to the invention, by a few scientists of vision, of their own peculiarly-adapted version of the lab notebook, and of the specialized management processes that produce it."
```

But, in the explicit table `wikize_refs.py` produces, it is still creating off page links that are not valid.

**Comment by markcmiller86** [comment]
@bernhold I believe #1129 combined with adjustments to the footnotes in the doc as per [this comment](https://github.com/betterscientificsoftware/bssw.io/pull/1121#issuecomment-955766497) will fix issues.

**Comment by markcmiller86** [comment]
> @bernhold I believe #1129 combined with adjustments to the footnotes in the doc as per [this comment](https://github.com/betterscientificsoftware/bssw.io/pull/1121#issuecomment-955766497) will fix issues.

FYI...I was not able to get this committed.

File: Articles/Blog/2021-12-HPCAndScienceProcessCulture.md [hunk 1]
```diff
@@ -84,11 +84,11 @@ So I guess the question that I've raised in this essay is: "Is Scientific HPC A
 Carlo Graziani is a computational scientist at Argonne National Laboratory. He started out as an astrophysicist and still dabbles in the stuff, as well as in FLASH simulations, but he mostly works on machine learning and uncertainty quantification (defined as "statistics funded by the US Department of Energy"), in applications that range from weather forecasting to plasma physics to COVID-19 modeling.
 
 
-[1]: The beginnings of open-science HPC as a sustained activity&mdash;as distinct from classified science HPC conducted at US DOE national labs&mdash;can probably be dated to DOE's ASC Alliance program in the late 1990s.
```
Comment (markcmiller86) [inline Articles/Blog/2021-12-HPCAndScienceProcessCulture.md]: To be clear...what you had here were **NOT** *footnotes*. They were *invalid* GFM link definitions. The change you just made here in this commit makes them *valid* GFM link definitions.

File: Articles/Blog/2021-12-HPCAndScienceProcessCulture.md [hunk 2]
```diff
@@ -77,7 +77,7 @@ What might a "mature" computational science center look like? I think the ASCI-c
...
 
-So I guess the question that I've raised in this essay is: "Is Scientific HPC A Mature Field Yet?" Well, Scientific HPC is about 25, at least dating from the ASC Alliance program, and that's about enough time to make a grown-up.  Most of the first generation of seniors, together with their ~~bad management~~ imperfect habits should have retired by now, and it's been long enough that I'm guessing plenty of younger people have been noticing how much could be gained by raising our game at the management level. We can't count on Moore's law to make up in new, faster simulations for the mislaid, corrupted, or uninterpretable data from older ones, and that fact should be giving night sweats to quite a few people who write proposals for supercomputer time allocations. So I actually think I'm going to meet Geoff again quite soon. I still owe him a beer from the time I...um, well, that's a long story, and come to think of it he probably never realized it was me...
```
Comment (markcmiller86) [inline Articles/Blog/2021-12-HPCAndScienceProcessCulture.md]: Just curios surrounding text in `~~` is the [GFM way](https://github.github.com/gfm/#strikethrough-extension-). Did that not work on rendered site?

**Comment by markcmiller86** [comment]
@bernhold can I be of any help here?

**Comment by bernhold** [comment]
Thanks @markcmiller86, but I don't have time to play with this further.  Need to move on.

**Comment by karbarz** [comment]
@bernhold 
I made some light edits 👍

**Pull request merged**